### PR TITLE
GATEWAY-3240 - collapse feature-direct branch

### PR DESCRIPTION
**Pull request by mlandis5am**
There are still some pending code review and cleanup items, however I should have addressed most of the little ones from ticket 3190 and pull 227.  

Note: Added changes from feature-direct for 3244

### CODE REVIEW

File: Product/Production/Common/Properties/pom.xml [hunk 1]
```diff
@@ -11,6 +11,9 @@
     <name>${project.artifactId}</name>
     <properties>
         <log4jDefaultLevel>INFO</log4jDefaultLevel>
+        <directAnchorStore>${env.NHINC_PROPERTIES_DIR}/../gateway.jks</directAnchorStore>
+        <directPublicCertStore>${env.NHINC_PROPERTIES_DIR}/../gateway.jks</directPublicCertStore>
+        <directPrivateCertStore>${env.NHINC_PROPERTIES_DIR}/../gateway.jks</directPrivateCertStore>
     </properties>
```
Comment (mlandis5am) [inline Product/Production/Common/Properties/pom.xml:17]: These new properties are replaced as part of resource filtering - the smtp.agent.config can't be loaded with relative paths.  In practice the direct configuration will require these be replaced with specific keystore(s) for direct which is what we're doing in our testing.

File: Product/Production/Common/Properties/Dev/smtp.agent.config.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<SmtpAgentConfig>
+   <Domains>        
+      <AnchorStore type="Uniform" storeType="KeyStore" file="${directAnchorStore}" filePass="changeit" privKeyPass="changeit"/>
```
Comment (mlandis5am) [inline Product/Production/Common/Properties/Dev/smtp.agent.config.xml:3]: See new properties in the pom...

File: Product/Production/Common/Properties/pom.xml (hunk 1 above)
Comment (bhumphrey) [inline Product/Production/Common/Properties/pom.xml:17]: I think the direct stores should be named like the following:  direct-public.jks, direct-private.jks and and direct-ca.jks. In otherwords don't use gateway

Comment (mlandis5am) [inline Product/Production/Common/Properties/pom.xml:17]: makes sense - that should be do-able and clearer

On Thu, Dec 13, 2012 at 12:21 PM, Brian Humphrey
notifications@github.comwrote:

> In Product/Production/Common/Properties/pom.xml:
> 
> > @@ -11,6 +11,9 @@
> >      <name>${project.artifactId}</name>
> >      <properties>
> >          <log4jDefaultLevel>INFO</log4jDefaultLevel>
> > -        <directAnchorStore>${env.NHINC_PROPERTIES_DIR}/../gateway.jks</directAnchorStore>
> > -        <directPublicCertStore>${env.NHINC_PROPERTIES_DIR}/../gateway.jks</directPublicCertStore>
> > -        <directPrivateCertStore>${env.NHINC_PROPERTIES_DIR}/../gateway.jks</directPrivateCertStore>
> >    </properties>
> 
> I think the direct stores should be named like the following:
> direct-public.jks, direct-private.jks and and direct-ca.jks. In otherwords
> don't use gateway
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/CONNECT-Solution/CONNECT/pull/250/files#r2410144.

## 

Michael Landis
Software Engineer
www.5amsolutions.com | Software for Life(TM)
phone: 866.526.6042 x774
email: mlandis@5amsolutions.com
skype: mlandis5am

**Pull request merged**